src/Tpay/Provider/TpayApiBankListProvider.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CommerceWeavers\SyliusTpayPlugin\Tpay\Provider;
+
+use CommerceWeavers\SyliusTpayPlugin\Payum\Exception\UnableToGetBankListException;
+use CommerceWeavers\SyliusTpayPlugin\Payum\Request\Api\GetTpayTransactionsChannels;
+use Payum\Core\Model\ArrayObject;
+use Payum\Core\Payum;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+final class TpayApiBankListProvider implements TpayApiBankListProviderInterface
+{
+    public function __construct(
+        private readonly Payum $payum,
+        private readonly CacheInterface $cache,
+    ) {
+    }
+
+    public function provide(): array
+    {
+        $gateway = $this->payum->getGateway('tpay');
+
+        /** @var GetTpayTransactionsChannels $value */
+        $value = $this->cache->get('tpay_bank_list', function (ItemInterface $item) use ($gateway): GetTpayTransactionsChannels {
+            $item->expiresAfter(new \DateInterval('P1D'));
+            $gateway->execute($value = new GetTpayTransactionsChannels(new ArrayObject()), true);
+
+            return $value;
+        });
```

Comment:
As caching can be considered as a separate responsibility, and it makes it easier to decide whether we want to get cachable/non-cachable service, it is a good practice to use the decorator pattern. Ref:
- https://github.com/CommerceWeavers/SyliusTpayPlugin/blob/main/src/Tpay/Security/Notification/Resolver/CachedCertificateResolver.php